async-nats/src/lib.rs

```diff
@@ -903,8 +907,42 @@ impl Subscriber {
     ///  subscriber.unsubscribe();
     /// # Ok(())
     /// # }
-    pub fn unsubscribe(self) {
-        drop(self)
+    pub async fn unsubscribe(&mut self) {
+        self.receiver.close()
```

Comment:
Again, would've expected unsubscribe to be sent.
Maybe `close` immediately closes the stream and `unsubscribe` does the unsubscribe route?

Reply:
I think that it's fine after last commit.

Not sure why we would like to `close()` the stream and don't do even async unsub. Looks more like a proto thing that we should take care of (and we do in `unsubscribe()`

Reply:
I think I would drop the sender, so that its all done centrally in the same place?

Reply:
If we want to handle unsubscribe by sending `ClientOp` and then dropping `Subscription` in `process`, we should  IMO add `oneshot` and wait for that drop before returning from `unsubscribe`. Otherwise you can still get messages after you called `unsubscribe`.